src/dune

```diff
@@ -1,10 +1,28 @@
+;; `melange.emit` is a Dune stanza that will produce build rules to generate
+;; JavaScript files from sources using the Melange compiler
+;; https://dune.readthedocs.io/en/stable/melange.html#melange-emit
 (melange.emit
+
+ ;; The `target` field is used by Dune to put all JavaScript artifacts in a specific folder inside `_build/default`
+ ;; This field `target` allows to have multiple `melange.emit` stanzas in the `dune` file
+ ;; if, for example, we want to produce ES6 and CommonJS output from the same sources.
```

Comment:
Could we include something that will
Make the output js files sit right next to the .re files? I’ve found the ‘promote until clean’ to be a bit weird, sometimes creating a new node_modules folder in my source files and stuff like that - perhaps I’m doing that wrong. An example would be super helpful 🙌

Reply:
I am not sure it is possible to produce js files right next to the re files right now. They would be inside a folder including the path to dune file + target in it.

E.g. if the dune file and sources are in `src/foo/`, and target is `bar`, the js files would be placed in `src/foo/bar/src/foo`. There is some more info in https://melange.re/v2.0.0/build-system/#javascript-artifacts-layout.

@anmonteiro [looked into](https://github.com/ocaml/dune/pull/6541) removing `target` but there are some limitations in dune at the moment that prevent it.

Besides that specific thing with paths, one can use promotion normally. I added an commented example in I am not sure it is possible to produce js files right next to the re files right now. They would be inside a folder including the path to dune file + target in it.

E.g. if the dune file and sources are in `src/foo/`, and target is `bar`, the js files would be placed in `src/foo/bar/src/foo`. There is some more info in https://melange.re/v2.0.0/build-system/#javascript-artifacts-layout.

@anmonteiro [looked into](https://github.com/ocaml/dune/pull/6541) removing `target` but there are some limitations in dune at the moment that prevent it, let me know if it helps!